extensions/libraries/objectrepository/objectrepository.lcb

```diff
@@ -0,0 +1,147 @@
+/*
+Copyright (C) 2017 LiveCode Ltd.
+
+This file is part of LiveCode.
+
+LiveCode is free software; you can redistribute it and/or modify it under
+the terms of the GNU General Public License v3 as published by the Free
+Software Foundation.
+
+LiveCode is distributed in the hope that it will be useful, but WITHOUT ANY
+WARRANTY; without even the implied warranty of MERCHANTABILITY or
+FITNESS FOR A PARTICULAR PURPOSE.  See the GNU General Public License
+for more details.
+
+You should have received a copy of the GNU General Public License
+along with LiveCode.  If not see <http://www.gnu.org/licenses/>.  */
+
+/**
+This library provides support persistent object references within a session
+*/
+
+library com.livecode.library.objectrepository
+
+use com.livecode.engine
+
+metadata version is "1.0.0"
+metadata author is "LiveCode"
+metadata title is "Object Repository Library"
+metadata userVisible is "false"
+
+variable mSequence as Integer
+
+variable mObjects as Array
+
+/**
+
+Store an object in the object repository and get its ID
+
+Description:
+As long ID of an object may mutate during a session the IDE needs an
+object reference that can handle that mutation.
+
+Parameters:
+pObject (String): An object long ID
+
+Returns:
+An object repository ID which can be used to get the current long id of
+the object
+
+*/
+public handler objectRepositoryStore(in pObject as String) returns String
+   variable tID as String
+   repeat for each key tID in mObjects
+      if mObjects[tID] exists then
+         get property "long id" of mObjects[tID]
+   		if the result is pObject then
+            return tID
+         end if
+      end if
+   end repeat
```

Comment:
This loop looks like it could be replaced with objectRepositoryRetrieve - with the added benefit of the object existence check in that function

Reply:
Hmm... `objectRepositoryRetrieve` gets the long id for an object given its repository ID while this loop works out if an object with a given long id is already stored and returns the repository ID. I don't think one could be replaced by the other. I do wonder if this loop would be better resolving the long ID `pObject` before the loop and comparing directly to `mObjects[tID]` though rather than getting the long id and doing string comparison. What do you think @livecodeian ?

Reply:
I think I meant calling that function on each ID in the array, but looking at it again, aside from removing non-extant objects from the array it won't make much difference.